fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -2243,6 +2292,29 @@ void MultiVersionApi::updateSupportedVersions() {
 	}
 }
 
+ThreadFuture<Void> MultiVersionApi::updateClusterSharedStateMap(std::string clusterFilePath, Reference<IDatabase> db) {
+	MutexHolder holder(lock);
+	if (clusterSharedStateMap.find(clusterFilePath) == clusterSharedStateMap.end()) {
```

Comment:
Do we need to increase the reference count for referencing it in the map? As I understand, if the shared object stays in the map after the last database instance is closed, then if we open a new database it may get a stale reference.

Reply:
This is a good point and makes me consider whether we can remove the reference counting altogether if it will always be present in the map until it is cleared. Originally the point of counting references was to avoid early cleanup and also avoid leaking the memory (missing the cleanup). In this case, I could just move the cleanup to `clearClusterSharedStateMapEntry`. What do you think?

Reply:
I implemented a rough sketch of the solution you proposed (not matching exactly), but based on your comment:

> So the objects dynamically allocated on one network thread must also be deleted by the same network thread.

I think this can be simplified much further by using the `onMainThreadVoid` function to facilitate the deletion, since the creation is also guaranteed to be done by the main thread as well. Let's discuss this further in our 1:1 if this is a viable solution.